python/tests/test_approximate_nearest_neighbors.py

```diff
@@ -1046,36 +1046,94 @@ def test_return_fewer_k(
         int64_max = np.iinfo("int64").max
         float_inf = float("inf")
 
-        # ensure consistency with cuvs for ivfflat, and ivfpq > 24.10
-        import cuvs
-        from packaging import version
-
-        if algorithm == "ivfflat" or version.parse(cuvs.__version__) > version.parse(
-            "24.10.00"
-        ):
-            ann_evaluator = ANNEvaluator(X, k, metric)
-            spark_indices = np.array([row["indices"] for row in knn_df_collect])
-            spark_distances = np.array([row["distances"] for row in knn_df_collect])
-            ann_evaluator.compare_with_cuml_or_cuvs_sg(
-                algorithm, algo_params, spark_indices, spark_distances, tolerance=0.0
-            )
-
         # check result details
+        # note that cuVS produces non-deterministic results during the indexing process
         indices_none_probed = [int64_max, int64_max, int64_max, int64_max]
         distances_none_probed = [float_inf, float_inf, float_inf, float_inf]
 
         def check_row_results(
-            i: int, indices_if_probed: List[int], distances_if_probed: List[float]
+            i: int,
+            indices_if_probed: List[int],
+            distances_if_probed: List[float],
+            res_indices: List[int],
```

Comment:
Looks like these types might be incorrect.   See below. List[List[int]]?

Reply:
Yeah, revised. Good catch!